### PR TITLE
Add tags for posts

### PR DESCRIPTION
**Pull request by gracelungu**
#### What does this PR do?
This PR adds the possibility to attach tags to posts

#### Description of Task to be completed?
- Add the tag dropdown component to the Postcard
- Add tags to the request
- Add tags to Feed cards
- Write the test

#### How should this be manually tested?
- Click the link below to open the application
- Login with social
- Make a post with tags

#### Screenshots (if appropriate)
image

image

image

### CODE REVIEW

**Comment by nkpremices** [comment]
@gracelungu The display looks fine to me. However, I would suggest to make the dropdown list scrollable and give it a max-height which might contain just 5 tags for ex.. It looks a little bit with too much height to me.

**Comment by gracelungu** [comment]
@nkpremices The dropdown is scrollable with a maximum height to make sure it always fit on the user's screen

**Comment by oesukam** [comment]
@gracelungu Shouldn't it be added on the modal to along with choosing the category?

File: src/components/PostTextArea/PostTextArea.jsx [hunk 1]
```diff
@@ -280,6 +281,10 @@ export class PostTextArea extends Component {
     );
   }
 
+  handleTagsSelected = (tags) => {
+    this.setState((state) => ({ ...state, tags }));
```
Comment (oesukam) [inline src/components/PostTextArea/PostTextArea.jsx]: Why using the callback?

File: src/components/TagsDropDown/index.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import TAGS from '../../constants/tags';
+import './index.scss';
+
+const TagsDropDown = ({ onTagsSelected }) => {
+  const { tick } = useSelector(({ posts }) => posts);
+
+  const [showTags, setShowTags] = useState();
```
Comment (oesukam) [inline src/components/TagsDropDown/index.jsx:10]: Are we mixing hooks with class component in the same code base?

Comment (gracelungu) [reply src/components/TagsDropDown/index.jsx:10]: @oesukam I find it better to write new components in hooks and progressively refactor existing components too, as hooks reduce complexity.

**Comment by gracelungu** [comment]
@oesukam Having the tags added on the TextArea component is much better than mixing the whole thing with categories in the modal, this way the user can always interact with the tags as an addon just like we have emojis and pictures before clicking on the post button.

**Comment by oesukam** [comment]
@gracelungu I didn't get me right. Not to the text area but to the modal before posting because you should also allow custom tags, not only predefined ones. Also listing all tags verticaly doesn't give a good user experience

File: src/components/TagsDropDown/index.jsx (hunk 2 above)
Comment (oesukam) [reply src/components/TagsDropDown/index.jsx:10]: OK. You can provide a default value instead of having it as null

**Comment by oesukam** [comment]
@gracelungu Any updates?

**Pull request merged**